clustered/server/src/main/java/org/ehcache/clustered/server/state/MessageTracker.java

```diff
@@ -79,17 +84,32 @@ public boolean isEmpty() {
     return nonContiguousMsgIds.isEmpty();
   }
 
+
+  /**
+   * Notify Message tracker that the sync is completed.
+   */
+  public void notifySyncCompleted() {
+    this.isSyncCompleted = true;
+  }
+
   /**
    * Remove the contiguous seen msgIds from the nonContiguousMsgIds and update highestContiguousMsgId
    */
   private void reconcile() {
 
     // This happens when a passive is started after Active has moved on and
     // passive starts to see msgIDs starting from a number > 0.
-    if (highestContiguousMsgId == -1L && nonContiguousMsgIds.size() > 100) {
-      Long min = nonContiguousMsgIds.first();
+    // Once the sync is completed, fast forward highestContiguousMsgId.
+    if (highestContiguousMsgId == -1L && isSyncCompleted) {
```

Comment:
Removed the number 100 check, as it is no longer required. Since after sync completion all future msgIds should arrive at passive.